src/clj/puppetlabs/services/jruby_pool_manager/impl/jruby_internal.clj

```diff
@@ -42,18 +46,38 @@
     :force RubyInstanceConfig$CompileMode/FORCE
     :off RubyInstanceConfig$CompileMode/OFF))
 
+(schema/defn ^:always-validate get-profiling-mode :- RubyInstanceConfig$ProfilingMode
+  [config-profiling-mode :- jruby-schemas/SupportedJRubyProfilingModes]
+  (case config-profiling-mode
+    :api RubyInstanceConfig$ProfilingMode/API
+    :flat RubyInstanceConfig$ProfilingMode/FLAT
+    :graph RubyInstanceConfig$ProfilingMode/GRAPH
+    :html RubyInstanceConfig$ProfilingMode/HTML
+    :json RubyInstanceConfig$ProfilingMode/JSON
+    :off RubyInstanceConfig$ProfilingMode/OFF
+    :service RubyInstanceConfig$ProfilingMode/SERVICE))
+
 (schema/defn ^:always-validate init-jruby :- jruby-schemas/ConfigurableJRuby
   "Applies configuration to a JRuby... thing.  See comments in `ConfigurableJRuby`
   schema for more details."
   [jruby :- jruby-schemas/ConfigurableJRuby
    config :- jruby-schemas/JRubyConfig]
-  (let [{:keys [ruby-load-path compile-mode lifecycle]} config
+  (let [{:keys [ruby-load-path compile-mode lifecycle profiling-mode profiler-output-file]} config
         initialize-scripting-container-fn (:initialize-scripting-container lifecycle)]
     (doto jruby
       (.setLoadPaths ruby-load-path)
       (.setCompileMode (get-compile-mode compile-mode)))
     (when-let [compat-version default-jruby-1-7-compat-version]
       (.setCompatVersion jruby compat-version))
+    (when (and profiler-output-file (not= :off profiling-mode))
```

Comment:
I would consider breaking this part out into its own function that takes a jruby and modifies it/logs as necessary? Just to keep this function simple

Reply:
seems good